### ISSUE TITLE
Unblocking adds to certain web pages but not for all pages

### ISSUE THREAD

**Issue by d4nielpage** [pi-hole/pi-hole#291]
Is there a way to whitelist certain pages i.e some web pages require that you deactivate your adblocker before they will work, but just adding that particular page to the white list dosent work as the ads within that page are still getting blocked, is it possible to bypass the adblock for certain pages so the ads will load but remain blocked for other pages. one page in particular that i can only access by changing my DNS back to google http://www.hackintosh.zone

**Comment by jacobsalmela** [comment]
Unfortunately, no.  Since Pi-hole works at the DNS level, it operates differently than most traditional ad blockers.  Take a look at [this FAQ](https://pi-hole.net/faq/how-do-i-whitelist-or-blacklist-a-webiste-or-domain/).  You need to determine what advertisers the Website is using.  I found these domains:
```
www.hackintosh.zone
cdn.hackintosh.zone
pagead2.googlesyndication.com
www.google-analytics.com
fonts.googleapis.com
googleads.g.doubleclick.net
fonts.gstatic.com
ajax.googleapis.com
ping.chartbeat.net
pi-hole.net
img.hackintosh.zone
cdn.goroost.com
www.hackintosh.computer
cdn.hackintosh.computer
```
It looks like Google Ads are being used, which means if you whitelisted that domain, you would also see ads on other sites that use Google Ads.  As of now, the Pi-hole blocks a domain as all-or-nothing.  We are actively developing and we will continue to look for a way around this.

**Comment by jobro13** [comment]
PiHole is now being used as a DNS-server.

A way I sense around this issue is to make PiHole return it's own IP for every request, so it can scan the packets sent and act accordingly. But on the other hand, that also seems a very messy approach, and I'm sure there are many security concerns too.

**Comment by PromoFaux** [comment]
Out of scope, but added archive label to find this later should we ever think of a way around it.
